src/services/suggestionDiagnostics.ts

```diff
@@ -109,7 +113,118 @@ namespace ts {
         }
     }
 
+    function addConvertToAsyncFunctionDiagnostics(node: Node, checker: TypeChecker, diags: DiagnosticWithLocation[]): void {
+        if (isAsyncFunction(node)) {
+            return;
+        }
+
+        if (isFunctionLikeDeclaration(node) && !node.body) {

```

Comment:
no need to check `isFunctionLikeDeclaration` again. you already asserted that before the call.

Reply:
I would also combine the 2 bailout statements into one:

```ts
if (!node.body || isAsyncFunction(node)) {
    return;
}
```